src/argilla/client/sdk/text_classification/api.py

```diff
@@ -58,6 +58,22 @@ def data(
         )
 
 
+def add_dataset_labeling_rules(
+    client: AuthenticatedClient, name: str, rules: List[LabelingRule]
+) -> Response[Union[HTTPValidationError, ErrorMessage]]:  # not sure if correct :)
+    url = "{}/api/datasets/TextClassification/{name}/labeling/add_rules".format(
+        client.base_url, name=name, data=rules
+    )  # probably wrong but hopefully close
+
+    response = httpx.post(
+        url=url,
+        headers=client.get_headers(),
+        cookies=client.get_cookies(),
+        timeout=client.get_timeout(),
+    )
+    return build_list_response(response)
+
+
```

Comment:
The API endpoint is as POST HTTP method to the path `/api/datasets/{name}/TextClassification/labeling/rules`.

You should create a method to create a single rule and use it in a loop from the `api.add_dataset_labeling_rules` method.

```suggestion
def add_dataset_labeling_rule(
    client: AuthenticatedClient,
    name: str,
    rule: LabelingRule,
) -> Response[
    Union[LabelingRule, HTTPValidationError, ErrorMessage]
]:
    url = "{}/api/datasets/{name}/TextClassification/labeling/rules".format(
        client.base_url,
        name=name,
        json=rule.dict(),
    )

    response = httpx.post(
        url=url,
        headers=client.get_headers(),
        cookies=client.get_cookies(),
        timeout=client.get_timeout(),
    )
    return build_typed_response(response, rule.__class__)
```